Fix: Draft post date change in quick edit

itzmekhokan opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

This PR fix draft post date change save issues in quick edit, considering ticket #59125 issue and previously ticket task #19907 done patch which create issues ( not updating date/time ) for post status - ( 'draft', 'pending', 'auto-draft' ).

Trac ticket: https://core.trac.wordpress.org/ticket/59125

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

itzmekhokan committed:
Fix: Draft post date change in quick edit

itzmekhokan committed:
Fix: Extra white space

joedolson commented on src/wp-admin/includes/post.php:
```diff
@@ -197,7 +193,22 @@ function _wp_translate_postdata( $update = false, $post_data = null ) {
+		 *
+		 * This changes is done considering ticket #59125 issue and previously ticket task #19907 done patch,
+		 * Which create issues ( not updating date/time ) for post status - ( 'draft', 'pending', 'auto-draft' ).
+		 */
```
Let's simplify this comment; I think we only need 

```
/*
 * Only assign a post date if the user has explicitly set a new value.
 * See #59125 and #19907.
 */
```

joedolson commented on src/wp-admin/includes/post.php:
```diff
@@ -197,7 +193,22 @@ function _wp_translate_postdata( $update = false, $post_data = null ) {
+			$post_data['post_date_gmt'] = get_gmt_from_date( $post_data['post_date'] );
+		} else {
+			$post_data['edit_date'] = false;
+			unset( $post_data['post_date'] );
```
Also need to move `unset( $post_data['post_date_gmt'] );` into this block; otherwise it's being set with an array value that may not exist.

joedolson reviewed (changes requested):
Also needs tests updates. Incoming.

joedolson committed:
add tests & update comments/code

joedolson committed:
Test needs to use existing post date as comparison.

joedolson committed:
Need to also set time values.

joedolson committed:
Adding an additional assertion

I'm not sure why this isn't working, and I hope this assertion will help.

Would sure be easier if phpunit tests were working locally...

joedolson committed:
Does not help to add to the wrong test case.

joedolson committed:
Invalid use of function

joedolson committed:
Restore to intended assertion.
